CompositionRoot/Factories/DFactory.swift

```diff
@@ -0,0 +1,16 @@
+protocol DFactoryProtocol {
+    func createD() -> DProtocol
+}
+
+class DFactory: DFactoryProtocol {
+    let applicationContext: ApplicationContext
+    
+    init(applicationContext: ApplicationContext) {
```

Comment:
Composition Root pattern transit the `ApplicationContext` from high level class to low level class through factory classes.  This free the business classes from transitioning dependency job.